tests/e2e/provisioning/test/test_suite.go

```diff
@@ -0,0 +1,75 @@
+package test
+
+import (
+	"crypto/tls"
+	"net/http"
+	"testing"
+
+	"github.com/kyma-incubator/compass/tests/e2e/provisioning/pkg/client/runtime"
+
+	"github.com/stretchr/testify/assert"
+
+	"github.com/google/uuid"
+	"github.com/kyma-incubator/compass/tests/e2e/provisioning/pkg/client/broker"
+	"github.com/kyma-incubator/compass/tests/e2e/provisioning/pkg/client/kyma"
+	"github.com/sirupsen/logrus"
+	"github.com/stretchr/testify/require"
+	"github.com/vrischmann/envconfig"
+)
+
+type Config struct {
+	Broker  broker.Config
+	Runtime runtime.Config
+
+	TenantID string `default:"3e64ebae-38b5-46a0-b1ed-9ccee153a0ae"`
+}
+
+// Suite provides set of clients able to provision and test Kyma runtime
+type Suite struct {
+	t *testing.T
+
+	log           logrus.FieldLogger
+	brokerClient  *broker.Client
+	kymaClient    *kyma.Client
+	runtimeClient *runtime.Client
+}
+
+func (ts *Suite) TearDown() {
+	ts.log.Info("Cleaning up...")
+	err := ts.runtimeClient.TearDown()
```

Comment:
hm.. `TearDown` on `runtimeClient` looks quite strange. Can we just say 
```
ts.runtimeClient.EnsureUAAInstanceRemoved
```

You already did sth similar for `ts.brokerClient.DeprovisionRuntime()`